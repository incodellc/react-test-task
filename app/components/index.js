export { PageNotFound } from './PageNotFound';
export { Header } from './Header';
export { TickerInfo } from './TickerInfo';
export { FetchIntervalSelector } from './FetchIntervalSelector';
