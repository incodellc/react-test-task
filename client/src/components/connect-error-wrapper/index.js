import ConnectErrorWrapper from './ConnectErrorWrapper';
export default ConnectErrorWrapper;