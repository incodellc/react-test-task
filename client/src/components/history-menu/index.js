import HistoryMenu from './HistoryMenu';
export default HistoryMenu;