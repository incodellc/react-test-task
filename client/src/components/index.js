import InfoBar from './info-bar';
import Layout from './layout';
import ConnectErrorWrapper from './connect-error-wrapper';
import TickerList from './ticker-list';
import TickerListItem from './ticker-list-item';

export { InfoBar, Layout, ConnectErrorWrapper, TickerList, TickerListItem };