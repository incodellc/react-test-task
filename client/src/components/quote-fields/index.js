import QuoteFields from './QuoteFields';
export default QuoteFields;