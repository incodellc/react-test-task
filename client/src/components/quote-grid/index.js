import QuotesGrid from './QuotesGrid';
export default QuotesGrid;