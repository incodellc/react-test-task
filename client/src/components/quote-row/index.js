import QuoteRow from './QuoteRow';
export default QuoteRow;