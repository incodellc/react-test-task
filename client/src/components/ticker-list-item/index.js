import TickerListItem from './TickerListItem';
export default TickerListItem;