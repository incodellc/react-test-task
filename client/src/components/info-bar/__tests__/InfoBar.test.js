import React from 'react';
import { InfoBar } from '../../../components';
import { cleanup, render } from '@testing-library/react';

describe('Check InfoBar component', () => {
    afterEach(cleanup);

    it('component renders correctly', () => {
        const { asFragment } = render(<InfoBar />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('component should have correct content', () => {
        const { getByText } = render(<InfoBar />);
        expect(getByText('Kirill Kopitsa')).toBeInTheDocument();
        expect(getByText('Dnipro, Ukraine')).toBeInTheDocument();
        expect(getByText('joseff.knecht@gmail.com')).toBeInTheDocument();
        expect(getByText('github.com/joseffone')).toBeInTheDocument();
        expect(getByText('This site is a solution of the test task from InCode Group.')).toBeInTheDocument();
    });
});