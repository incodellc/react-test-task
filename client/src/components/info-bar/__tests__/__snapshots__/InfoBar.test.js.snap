// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check InfoBar component component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ui segment ui left sidebar infoBar"
    data-testid="info-bar"
  >
    <div
      class="ui container"
    >
      <div
        class="ui stackable grid"
      >
        <div
          class="three wide column"
        />
        <div
          class="five wide column"
        >
          <h4
            class="ui header"
          >
            Contacts
          </h4>
          <div
            class="ui list"
            role="list"
          >
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="user icon"
              />
              <div
                class="content"
              >
                Kirill Kopitsa
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="marker icon"
              />
              <div
                class="content"
              >
                Dnipro, Ukraine
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="mail icon"
              />
              <div
                class="content"
              >
                <a
                  href="mailto:joseff.knecht@gmail.com"
                >
                  joseff.knecht@gmail.com
                </a>
              </div>
            </div>
            <div
              class="item"
              role="listitem"
            >
              <i
                aria-hidden="true"
                class="github icon"
              />
              <div
                class="content"
              >
                <a
                  href="https://github.com/joseffone"
                >
                  github.com/joseffone
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="five wide column"
        >
          <h4
            class="ui header"
          >
            About project
          </h4>
          <p>
            This site is a solution of the test task from InCode Group.
          </p>
        </div>
        <div
          class="three wide column"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
